app/models/notification.rb

```diff
@@ -28,20 +29,29 @@ class Notification < ApplicationRecord
 
   def localized_message
     case purpose
-    when "missed_visit_reminder", "experimental_appointment_reminder"
+    when "covid_medication_reminder"
       I18n.t(
         message,
-        facility_name: subject.facility.name,
+        facility_name: patient.assigned_facility.name,
         patient_name: patient.full_name,
-        appointment_date: subject.scheduled_date,
-        locale: subject.facility.locale
+        locale: patient.assigned_facility.locale
       )
-    when "covid_medication_reminder"
+    when "experimental_appointment_reminder"
+      facility = subject&.facility || patient.assigned_facility
       I18n.t(
         message,
-        facility_name: patient.assigned_facility.name,
+        facility_name: facility.name,
         patient_name: patient.full_name,
-        locale: patient.assigned_facility.locale
+        appointment_date: subject&.scheduled_date,
+        locale: facility.locale
```

Comment:
This is an edge case, but I'd argue that all of these locales should be `patient.locale`, which delegates to `patient.assigned_facility.locale`. If a patient ever had an appointment at a facility other than their assigned facility, they would get notifications in the usual language.

Reply:
Discussed this while on zoom w/ Kris, and deciding to ship this as is -- this sounds like a question more for Praveen about how this really should work.